### ISSUE TITLE
Can not get `scan` event for login QR Code

### ISSUE THREAD

**Issue by huan** [botorange/wechaty-puppet-padpro#120]
## 1. Versions
- What is your wechaty version?
Answer: 0.22.6

- What is your wechaty-puppet-padpro version?
Answer: 0.2.0

- What is your node version? (run `node --version`)
Answer: 11.10.1

- What os are you using
Answer: Ubuntu 18.04

## 2. Describe the bug
Give a clear and concise description of what the bug is.

After Wechaty start, there's no `scan` event emitted.

## 3. To Reproduce

**This part is very important: if you can not provide any reproduce steps, then the problem will be very hard to be recognized.**

1. clone https://github.com/wechaty/wechaty-getting-started
2. npm i
3. WECHATY_LOG=silly WECHATY_PUPPET_PADPRO_TOKEN=puppet_padpro_xxxx WECHATY_PUPPET=wechaty-puppet-padpro npm start

## 4. Expected behavior
Give a clear and concise description of what you expected to happen.

Get a `scan` event and show a QR Code

## 5. Actual behavior
If applicable, add screenshots to help explain your problem. But do not paste log screenshots here. (**Pleas do not just paste a log image**)

No `scan` event at all

## 6. Full Output Logs
Set env `WECHATY_LOG=silly` in order to set log level to silly, then we can get the full log (If you dosen't set log env, log level is info as default, we cannot get the full log)

**We need full log instead of log screenshot or log fragments!**

<details>
<summary>
Show Logs
</summary>

```shell
┌ zixia@zixia-desktop:~/chatie/wechaty-getting-started [18:45:03] tty:[5] jobs:[0]
└ (master) $ npm start

> wechaty-getting-started@0.1.0 start /home/zixia/chatie/wechaty-getting-started
> node examples/starter-bot.js

18:45:08 INFO Wechaty <default> start() v0.22.6 is starting...
18:45:08 INFO PuppetManager install(wechaty-puppet-puppeteer@^0.14.1) please wait ...
npm: 
> puppeteer@1.15.0 install /home/zixia/chatie/wechaty-getting-started/node_modules/wechaty/node_modules/puppeteer
> node install.js


npm: 

npm: Chromium downloaded to /home/zixia/chatie/wechaty-getting-started/node_modules/wechaty/node_modules/puppeteer/.local-chromium/linux-650583

npm: npm
npm:  notice created a lockfile as package-lock.json. You should commit this file.

npm: npm 
npm: WARN wechaty-puppet-puppeteer@0.14.6 requires a peer of file-box@^0.8.22 but none is installed. You must install peer dependencies yourself.
npm WARN wechaty-puppet-puppeteer@0.14.6 requires a peer of wechaty-puppet@^0.14.1 but none is installed. You must install peer dependencies yourself.

npm: npm WARN rx-queue@0.6.2 requires a peer of brolog@^1.4.9 but none is installed. You must install peer dependencies yourself.

npm: 

npm: + wechaty-puppet-puppeteer@0.14.6
added 140 packages from 157 contributors and audited 292 packages in 45.911s

npm: found 0 vulnerabilities


18:45:54 INFO PuppetManager install(wechaty-puppet-puppeteer@^0.14.1) done
Starter Bot Started.
▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
█ ▄▄▄▄▄ █▄▀ ▀ ▄█▄▄▀█▄▄█ ▄▄▄▄▄ █
█ █   █ █   █▀ ▀▄█▀ ▄▀█ █   █ █
█ █▄▄▄█ █▄█▀ ▄▀▀█▄▀▀ ██ █▄▄▄█ █
█▄▄▄▄▄▄▄█▄█ █ █ █▄▀▄▀▄█▄▄▄▄▄▄▄█
█ ▄▄ █▀▄▀ █ ▄  ▄▀▀▄ ▀█ ▀▀ ▄▄█ █
█▀ █  ▀▄▀▀  ▄▄▀ ▀ █▀█ ▄▄█▀█▄▄▀█
█▀▀▄▄  ▄ ▄▄▄▀ █▄  ▄█ █▄▀██▄ ▀▀█
█▄▀█ ▀█▄█ ██ █▄▀▀▄▄▄▄▄ ▀  █▄ ▄█
█▄▀▄ ▄ ▄ ▀█▀█▄▄   ▀▄▄  █ ▄ ▀ ██
█▄█ █▀█▄▀ ▄▄▀██▄▄▄▀▀█▀▀▀▄▀█ ███
██▄█▄█▄▄▄ ▄▄██▀▀ ▀ ▀▀ ▄▄▄   ▀▀█
█ ▄▄▄▄▄ ███ ▀▀▀▀▀▄█ ▄ █▄█  █▄██
█ █   █ █▀▀  █▀▀  ▄█▀  ▄  ▀▄▄ █
█ █▄▄▄█ █ ▀▀█ ▀▄█▀█ ▄ ██▄▀▀▄▀▄█
█▄▄▄▄▄▄▄█▄▄█▄███▄███▄▄▄██▄██▄██

https://api.qrserver.com/v1/create-qr-code/?data=https%3A%2F%2Flogin.weixin.qq.com%2Fl%2FwY8sQNLtIQ%3D%3D
^C
┌ zixia@zixia-desktop:~/chatie/wechaty-getting-started [18:45:59] tty:[5] jobs:[0]
└ (master) $ WECHATY_LOG=silly WECHATY_PUPPET_PADPRO_TOKEN=puppet_padpro_xxxxxxxx WECHATY_PUPPET=wechaty-puppet-padpro npm start

> wechaty-getting-started@0.1.0 start /home/zixia/chatie/wechaty-getting-started
> node examples/starter-bot.js

18:46:12 SILL Config WECHATY_LOG set level to silly
18:46:12 INFO Config registering process.on("unhandledRejection") for development/debug
18:46:12 VERB Config constructor()
18:46:12 SILL Accessory #0<Wechaty> constructor()
18:46:12 VERB Wechaty contructor()
18:46:12 VERB StateSwitch constructor(name=Wechaty)
18:46:12 VERB StateSwitch constructor(name=WechatyReady)
18:46:12 VERB Wechaty on(scan, function) registered
18:46:12 VERB Wechaty onFunction(scan)
18:46:12 VERB Wechaty on(login, function) registered
18:46:12 VERB Wechaty onFunction(login)
18:46:12 VERB Wechaty on(logout, function) registered
18:46:12 VERB Wechaty onFunction(logout)
18:46:12 VERB Wechaty on(message, function) registered
18:46:12 VERB Wechaty onFunction(message)
18:46:12 SILL Wechaty version() form development environment is not availble: ENOENT: no such file or directory, stat '/home/zixia/chatie/wechaty-getting-started/node_modules/wechaty/dist/.git'
18:46:12 INFO Wechaty <wechaty-puppet-padpro> start() v0.22.6 is starting...
18:46:12 VERB Wechaty puppet: undefined
18:46:12 VERB Wechaty profile: undefined
18:46:12 VERB Wechaty id: cjvjdwcwq00000f31fp3255jp
18:46:12 SILL StateSwitch <Wechaty> on() is false
18:46:12 SILL StateSwitch <WechatyReady> off() is true
18:46:12 VERB StateSwitch <WechatyReady> off(true) <- (true)
18:46:12 SILL StateSwitch <Wechaty> on() is false
18:46:12 VERB StateSwitch <Wechaty> on(pending) <- (false)
18:46:12 VERB MemoryCard constructor(undefined)
18:46:12 VERB MemoryCard getStorage() for storage type: N/A
18:46:12 VERB MemoryCard load() from storage: N/A
18:46:12 VERB MemoryCard load() no storage
18:46:12 VERB Wechaty initPuppet() 
18:46:12 VERB MemoryCard multiplex(puppet)
18:46:12 VERB MemoryCard static multiplex(MemoryCard<>, puppet)
18:46:12 VERB MemoryCard constructor({"multiplex":{"name":"puppet","parent":{"payload":{},"multiplexNameList":[]}}})
18:46:12 VERB PuppetManager resolve({wechaty: Wechaty#cjvjdwcwq00000f31fp3255jp<>(), puppet: wechaty-puppet-padpro, puppetOptions: undefined})
18:46:12 VERB PuppetManager resolveName(wechaty-puppet-padpro)
18:46:12 VERB PuppetManager checkModule(wechaty-puppet-padpro)
18:46:12 INFO PuppetManager install(wechaty-puppet-padpro@latest) please wait ...
npm: 
> nosql-leveldb@3.0.1 install /home/zixia/chatie/wechaty-getting-started/node_modules/wechaty/node_modules/nosql-leveldb
> prebuild-install || node-gyp rebuild


npm: prebuild-install
npm:  WARN install No prebuilt binaries found (target=11.10.1 runtime=node arch=x64 platform=linux)

npm: make: Entering directory '/home/zixia/chatie/wechaty-getting-started/node_modules/wechaty/node_modules/nosql-leveldb/build'

npm:   CXX(target) Release/obj.target/leveldb/deps/leveldb/leveldb-1.20/db/builder.o

npm:   CXX(target) Release/obj.target/leveldb/deps/leveldb/leveldb-1.20/db/db_impl.o

npm:   CXX(target) Release/obj.target/leveldb/deps/leveldb/leveldb-1.20/db/db_iter.o

npm:   CXX(target) Release/obj.target/leveldb/deps/leveldb/leveldb-1.20/db/filename.o

npm:   CXX(target) Release/obj.target/leveldb/deps/leveldb/leveldb-1.20/db/dbformat.o

npm:   CXX(target) Release/obj.target/leveldb/deps/leveldb/leveldb-1.20/db/log_reader.o

npm:   CXX(target) Release/obj.target/leveldb/deps/leveldb/leveldb-1.20/db/log_writer.o

npm:   CXX(target) Release/obj.target/leveldb/deps/leveldb/leveldb-1.20/db/memtable.o

npm:   CXX(target) Release/obj.target/leveldb/deps/leveldb/leveldb-1.20/db/repair.o

...

In file included from ../src/leveldown.cc:9:0:
../src/leveldown.h:54:51: warning: ‘v8::Local<v8::Object> v8::Value::ToObject() const’ is deprecated: Use maybe version [-Wdeprecated-declarations]
       && node::Buffer::HasInstance(from->ToObject())) {
                                                   ^
In file included from /home/zixia/.node-gyp/11.10.1/include/node/node.h:63:0,
                 from ../src/leveldown.cc:6:
/home/zixia/.node-gyp/11.10.1/include/node/v8.h:10254:15: note: declared here
 Local<Object> Value::ToObject() const {
               ^~~~~

npm: In file included from ../src/leveldown.cc:9:0:
../src/leveldown.h:55:47: warning: ‘v8::Local<v8::Object> v8::Value::ToObject() const’ is deprecated: Use maybe version [-Wdeprecated-declarations]
     sz_ = node::Buffer::Length(from->ToObject());
                                               ^
In file included from /home/zixia/.node-gyp/11.10.1/include/node/node.h:63:0,
                 from ../src/leveldown.cc:6:
/home/zixia/.node-gyp/11.10.1/include/node/v8.h:10254:15: note: declared here
 Local<Object> Value::ToObject() const {
               ^~~~~

npm: In file included from ../src/leveldown.cc:9:0:
../src/leveldown.h:56:45: warning: ‘v8::Local<v8::Object> v8::Value::ToObject() const’ is deprecated: Use maybe version [-Wdeprecated-declarations]
     ch_ = node::Buffer::Data(from->ToObject());
                                             ^
In file included from /home/zixia/.node-gyp/11.10.1/include/node/node.h:63:0,
                 from ../src/leveldown.cc:6:
/home/zixia/.node-gyp/11.10.1/include/node/v8.h:10254:15: note: declared here
 Local<Object> Value::ToObject() const {
               ^~~~~

/home/zixia/.node-gyp/11.10.1/include/node/v8config.h:326:3: note: in definition of macro ‘V8_DEPRECATED’
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~

npm:   SOLINK_MODULE(target) Release/obj.target/leveldown.node

npm:   COPY Release/leveldown.node

npm: make: Leaving directory '/home/zixia/chatie/wechaty-getting-started/node_modules/wechaty/node_modules/nosql-leveldb/build'

npm: 
> grpc@1.20.3 install /home/zixia/chatie/wechaty-getting-started/node_modules/wechaty/node_modules/grpc
> node-pre-gyp install --fallback-to-build --library=static_library


npm: node-pre-gyp
npm:  
npm: WARN Using request for node-pre-gyp https download 

npm: [grpc] Success: "/home/zixia/chatie/wechaty-getting-started/node_modules/wechaty/node_modules/grpc/src/node/extension_binary/node-v67-linux-x64-glibc/grpc_node.node" is installed via remote

npm: 
> fast-xml-parser@3.12.16 postinstall /home/zixia/chatie/wechaty-getting-started/node_modules/wechaty/node_modules/fast-xml-parser
> node tasks/postinstall.js || exit 0


npm: npm WARN rx-queue@0.6.2 requires a peer of brolog@^1.4.9 but none is installed. You must install peer dependencies yourself.

npm: npm WARN wechaty-puppet-puppeteer@0.14.6 requires a peer of file-box@^0.8.22 but none is installed. You must install peer dependencies yourself.
npm WARN wechaty-puppet-puppeteer@0.14.6 requires a peer of wechaty-puppet@^0.14.1 but none is installed. You must install peer dependencies yourself.
npm WARN wechaty-puppet-padpro@0.2.0 requires a peer of file-box@^0.8.27 but none is installed. You must install peer dependencies yourself.
npm WARN
npm:  wechaty-puppet-padpro@0.2.0 requires a peer of wechaty-puppet@^0.15.1 but none is installed. You must install peer dependencies yourself.
npm WARN rx-queue@0.8.1 requires a peer of brolog@^1.4.9 but none is installed. You must install peer dependencies yourself.

npm: 

npm: + wechaty-puppet-padpro@0.2.0
added 360 packages from 180 contributors and audited 1230 packages in 55.899s

npm: found 0 vulnerabilities


18:47:08 INFO PuppetManager install(wechaty-puppet-padpro@latest) done
18:47:08 VERB Puppet constructor({"timeout":240}) #0
18:47:08 VERB StateSwitch constructor(name=PuppetPadpro)
18:47:08 VERB MemoryCard constructor(undefined)
18:47:08 VERB MemoryCard getStorage() for storage type: N/A
18:47:08 VERB MemoryCard load() from storage: N/A
18:47:08 VERB MemoryCard load() no storage
18:47:08 VERB Puppet constructor() watchdog timeout set to 240 seconds
18:47:08 VERB HotImport callerResolve(., /home/zixia/chatie/wechaty-getting-started/node_modules/wechaty-puppet/dist/src/puppet.js)
18:47:08 SILL HotImport callerResolve() callsites() file=/home/zixia/chatie/wechaty-getting-started/node_modules/hot-import/dist/src/hot-import.js, type=Object
18:47:08 SILL HotImport callerResolve() callsites() file=/home/zixia/chatie/wechaty-getting-started/node_modules/wechaty-puppet/dist/src/puppet.js, type=null
18:47:08 SILL HotImport callerResolve() callsites() file=/home/zixia/chatie/wechaty-getting-started/node_modules/wechaty/node_modules/wechaty-puppet-padpro/dist/src/puppet-padpro.js, type=null
18:47:08 SILL HotImport callerResolve() callerFile=/home/zixia/chatie/wechaty-getting-started/node_modules/wechaty/node_modules/wechaty-puppet-padpro/dist/src/puppet-padpro.js
18:47:08 VERB Puppet constructor() childClassPath=/home/zixia/chatie/wechaty-getting-started/node_modules/wechaty/node_modules/wechaty-puppet-padpro/dist/src
18:47:08 VERB Puppet constructor() memory.load() done
18:47:08 VERB Puppet setMemory()
18:47:08 VERB Wechaty initPuppetEventBridge(Puppet#0<PuppetPadpro>()/PuppetPadpro#0)
18:47:08 VERB Wechaty initPuppetEventBridge() puppet.on(friendship) registered
18:47:08 VERB Wechaty initPuppetEventBridge() puppet.on(login) registered
18:47:08 VERB Wechaty initPuppetEventBridge() puppet.on(logout) registered
18:47:08 VERB Wechaty initPuppetEventBridge() puppet.on(message) registered
18:47:08 VERB Wechaty initPuppetEventBridge() puppet.on(room-invite) registered
18:47:08 VERB Wechaty initPuppetEventBridge() puppet.on(room-join) registered
18:47:08 VERB Wechaty initPuppetEventBridge() puppet.on(room-leave) registered
18:47:08 VERB Wechaty initPuppetEventBridge() puppet.on(room-topic) registered
18:47:08 VERB Wechaty initPuppetEventBridge() puppet.on(scan) registered
18:47:08 VERB Wechaty initPuppetEventBridge() puppet.on(dong) registered
18:47:08 VERB Wechaty initPuppetEventBridge() puppet.on(error) registered
18:47:08 VERB Wechaty initPuppetEventBridge() puppet.on(ready) registered
18:47:08 VERB Wechaty initPuppetEventBridge() puppet.on(reset) registered
18:47:08 VERB Wechaty initPuppetEventBridge() puppet.on(watchdog) registered
18:47:08 VERB Wechaty initAccessory(Puppet#0<PuppetPadpro>()/PuppetPadpro#0)
18:47:08 SILL Accessory <Contact> static set wechaty = "Wechaty#cjvjdwcwq00000f31fp3255jp<>()"
18:47:08 SILL Accessory <ContactSelf> static set wechaty = "Wechaty#cjvjdwcwq00000f31fp3255jp<>()"
18:47:08 SILL Accessory <Friendship> static set wechaty = "Wechaty#cjvjdwcwq00000f31fp3255jp<>()"
18:47:08 SILL Accessory <Message> static set wechaty = "Wechaty#cjvjdwcwq00000f31fp3255jp<>()"
18:47:08 SILL Accessory <Room> static set wechaty = "Wechaty#cjvjdwcwq00000f31fp3255jp<>()"
18:47:08 SILL Accessory <RoomInvitation> static set wechaty = "Wechaty#cjvjdwcwq00000f31fp3255jp<>()"
18:47:08 SILL Accessory <Contact> static set puppet = "Puppet#0<PuppetPadpro>()/PuppetPadpro#0"
18:47:08 SILL Accessory <ContactSelf> static set puppet = "Puppet#0<PuppetPadpro>()/PuppetPadpro#0"
18:47:08 SILL Accessory <Friendship> static set puppet = "Puppet#0<PuppetPadpro>()/PuppetPadpro#0"
18:47:08 SILL Accessory <Message> static set puppet = "Puppet#0<PuppetPadpro>()/PuppetPadpro#0"
18:47:08 SILL Accessory <Room> static set puppet = "Puppet#0<PuppetPadpro>()/PuppetPadpro#0"
18:47:08 SILL Accessory <RoomInvitation> static set puppet = "Puppet#0<PuppetPadpro>()/PuppetPadpro#0"
18:47:08 SILL Accessory <Wechaty> set puppet = "Puppet#0<PuppetPadpro>()/PuppetPadpro#0"
18:47:08 SILL StateSwitch <PuppetPadpro> on() is false
18:47:08 SILL StateSwitch <PuppetPadpro> on() is false
18:47:08 VERB StateSwitch <PuppetPadpro> on(pending) <- (false)
18:47:08 INFO WechatGateway init(puppet_padpro_10934a6fa8d23504, padpro.botorange.com:6666, undefined)
18:47:08 SILL StateSwitch <PuppetPadpro> off() is false 
```

</details>

## 7. Additional context

I'm at home behide a NAT router.

[bug]

**Comment by windmemory** [comment]
Please update `wechaty-puppet-padpro` `v0.2.1`. Should be fixed in that version. 

This is a bug, and fixed in `v0.2.1`

**Comment by windmemory** [comment]
Close this issue for now since no comments in last 14 days.
